config/config.go

```diff
@@ -378,6 +380,9 @@ func (c *DNSSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if len(c.Names) == 0 {
 		return fmt.Errorf("DNS-SD config must contain at least one SRV record name")
 	}
+	if strings.ToUpper(c.Type) != "SRV" && c.Port == 0 {
+		return fmt.Errorf("Port is required for all types except SRV records")
```

Comment:
Please start errors lower-cased. Also, some more context would be nice here. How about:

"a port is required in DNS-SD configs for all record types except SRV"